### PR TITLE
Add seccomp profiles to all pod specs created via reconciling

### LINKED ISSUE #9052: Add SeccompProfile to control plane components in Seed cluster
### What would you like to be added? / User Story

While user cluster components have all been updated with a seccomp profile to reduce the amount of allowed syscalls and improve isolation, control plane components in the seed cluster haven't been updated to do so as well to secure the Seed itself.

### Acceptance criteria

- All control plane components that support it run with seccompProfile=runtimeDefault

### PR DESCRIPTION
**Pull request by embik**
**What does this PR do / Why do we need it**:

Adds seccomp profiles to every PodSpec that gets defaulted during a reconcile. Also adds a test to check that all control plane components are using the seccomp profile.

**Does this PR close any issues?**:
Fixes #9052

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
All pods created by KKP are assigned the `RuntimeDefault` seccomp profile
```

Signed-off-by: Marvin Beckers <marvin@kubermatic.com>

### CODE REVIEW

**Comment by kron4eg** [comment]
@embik it looks like you want to use https://github.com/kubermatic/kubermatic/blob/5108893a10277e75563c75dc186520639fd418e7/pkg/resources/reconciling/wrapper.go#L113 but you didn't knew about its existence

**Comment by embik** [comment]
> @embik it looks like you want to use
> 
> https://github.com/kubermatic/kubermatic/blob/5108893a10277e75563c75dc186520639fd418e7/pkg/resources/reconciling/wrapper.go#L113
> 
> but you didn't knew about its existence

I was thinking of using that, but this is not used for a lot of things (for example, image pull secrets are probably injected into everything as well manually) - Do you think we should add the seccomp profile there instead?

**Comment by kron4eg** [comment]
> Do you think we should add the seccomp profile there instead?

Yeah, if we are serious about having seccomp by default always and to not forget it somewhere in the future.

**Comment by embik** [comment]
Good point. Let me rework the PR then.

File: pkg/resources/reconciling/wrapper.go
Comment (kron4eg) [inline pkg/resources/reconciling/wrapper.go]:
```suggestion
	if newPodSpec.SecurityContext.SeccompProfile == nil {
```

checking here `newPodSpec.SecurityContext` for nilness is useless since you just initialized it above

Comment (embik) [reply pkg/resources/reconciling/wrapper.go]: Good point. I initially misunderstood the function and was checking `oldPodSpec`, but now it does not make sense anymore to check for `nil`.

**Comment by embik** [comment]
/hold

I want CI to run against this, but I have not finished adjusting code comments to reflect on this yet. So the code itself is ready for being reviewed, but I'll work on properly documenting this before merging it.

**Review by kron4eg — APPROVED** [review]
/approve

**Review by kron4eg — APPROVED** [review]

**Pull request merged**
